src/main/java/com/google/devtools/build/lib/runtime/CriticalPathComponent.java

```diff
@@ -185,7 +188,10 @@ public int getId() {
    * the longestPhaseSpawnRunnerName to the longest running spawn runner name across all phases if
    * it exists.
    */
-  void addSpawnResult(SpawnMetrics metrics, @Nullable String runnerName) {
+  void addSpawnResult(SpawnMetrics metrics, @Nullable String runnerName, boolean wasRemote) {
+    if (wasRemote) {
+      this.remote = true;
+    }
```

Comment:
this.remote = wasRemote?

Reply:
it would be more accurate to say `this.remote = wasRemote || this.remote;`: I don't want a subsequent local spawn (which seems to be a thing now with discoverInputs) to reset the flag for a component (i.e. action) which had a previous remote spawn result. This is a 'flip high, ignore low' switch.

Reply:
Consider adding a comment that remote about this.

Reply:
Done